test/apps/iodemo/io_demo.cc

```diff
@@ -1923,58 +1941,92 @@ class DemoClient : public P2pDemoCommon {
-            log << io_op_names[op_id] << " " << throughput_mbs << " MBs";
-            total_bytes[op_id] = 0;
-
-            // Collect min/max among all connections
-            long delta_min = std::numeric_limits<long>::max(), delta_max = 0;
-            size_t min_index = 0, total_completed = 0;
-            for (size_t server_index = 0; server_index < _server_info.size();
-                 ++server_index) {
-                server_info_t& server_info = _server_info[server_index];
-                long delta_completed       = server_info.num_completed[op_id] -
-                                             server_info.prev_completed[op_id];
-                if ((delta_completed < delta_min) ||
-                    ((delta_completed == delta_min) &&
+        // Collect min/max among all connections
+        for (size_t server_index = 0; server_index < _server_info.size();
+             ++server_index) {
+            server_info_t& server_info = _server_info[server_index];
+
+            long min_max = 0, max = 0;
+            for (int i = 0; i < opts().operations.size(); ++i) {
+                io_op_t op           = opts().operations[i];
+                size_t min_index     = io_op_perf_info[op].min_index;
+                long delta_completed = get_num_completed_round(server_info, op);
+
+                if ((delta_completed < io_op_perf_info[op].min) ||
+                    ((delta_completed == io_op_perf_info[op].min) &&
                      (server_info.retry_count >
-                      _server_info[min_index].retry_count))) {
-                    min_index = server_index;
+                              _server_info[min_index].retry_count))) {
+                    io_op_perf_info[op].min_index = server_index;
                 }
 
-                delta_min        = std::min(delta_completed, delta_min);
-                delta_max        = std::max(delta_completed, delta_max);
-                total_completed += delta_completed;
+                io_op_perf_info[op].min           =
+                        std::min(delta_completed, io_op_perf_info[op].min);
+                io_op_perf_info[op].max           =
+                        std::max(delta_completed, io_op_perf_info[op].max);
+                io_op_perf_info[op].total        += delta_completed;
+                io_op_perf_info[IO_OP_MAX].total += delta_completed;
+                server_info.prev_completed[op]    =
+                        server_info.num_completed[op];
+
+                min_max = std::max(min_max, delta_completed);
+                max     = std::max(max, delta_completed);
+            }
+
+            // Find global minimum and maximum among all operations
+            io_op_perf_info[IO_OP_MAX].min =
+                    std::min(io_op_perf_info[IO_OP_MAX].min, min_max);
+            io_op_perf_info[IO_OP_MAX].max =
+                    std::max(io_op_perf_info[IO_OP_MAX].max, max);
```

Comment:
what if instead of special handling for IO_OP_MAX here, get_num_completed_round() would also support IO_OP_MAX?

Reply:
it will require adding new array element with index `IO_OP_MAX` for each server_info.
and also, we will need to calculate `min_max` not once, but every time some operations completed

Reply:
why? when getting OP_MAX it will return sum of both READ and WRITE counters

Reply:
> why? when getting OP_MAX it will return sum of both READ and WRITE counters

yes, we can calculate the sum of READ/WRITE instead of maximum of minimums. It is more simple.
but I think it's not worth adding special element for `IO_OP_MAX` - better calculate here before reporting total performance information.

Reply:
done